crates/isograph_cli/src/generate_artifacts.rs

```diff
@@ -64,16 +64,116 @@ pub(crate) fn generate_and_write_artifacts(
     Ok(artifact_count)
 }
 
+fn build_iso_overload_for_entrypoint<'schema>(
+    resolver_field_id: ResolverFieldId,
+    schema: &'schema ValidatedSchema,
+) -> (String, String) {
+    let resolver = schema.resolver(resolver_field_id);
+    let mut s: String = "".to_string();
+    let import = format!(
+        "import entrypoint_{} from '../__isograph/{}/{}/entrypoint.ts'\n",
+        resolver.type_and_field.underscore_separated(),
+        resolver.type_and_field.type_name,
+        resolver.type_and_field.field_name,
+    );
+    let formatted_field = format!(
+        "entrypoint {}.{}",
+        resolver.type_and_field.type_name, resolver.type_and_field.field_name
+    );
+    s.push_str(&format!(
+        "export function iso<T>(
+            param: T & MatchesWhitespaceAndString<'{}', T>
+        ): IdentityWithParam<typeof entrypoint_{}>;\n",
+        formatted_field,
+        resolver.type_and_field.underscore_separated(),
+    ));
+    (import, s)
+}
+
+fn build_iso_overload_for_schema_resolver(resolver: &ValidatedSchemaResolver) -> (String, String) {
+    let mut s: String = "".to_string();
+    let import = format!(
+        "import {{ ResolverParameterType as field_{} }} from './{}/{}/reader.ts'\n",
+        resolver.type_and_field.underscore_separated(),
+        resolver.type_and_field.type_name,
+        resolver.type_and_field.field_name,
+    );
+    let formatted_field = format!(
+        "field {}.{}",
+        resolver.type_and_field.type_name, resolver.type_and_field.field_name
+    );
+    s.push_str(&format!(
+        "export function iso<T>(
+            param: T & MatchesWhitespaceAndString<'{}', T>
+        ): IdentityWithParam<field_{}>;\n",
+        formatted_field,
+        resolver.type_and_field.underscore_separated(),
+    ));
+    (import, s)
+}
+
+fn build_iso_overload<'schema>(schema: &'schema ValidatedSchema) -> PathAndContent {
+    let mut imports = "".to_string();
+    let mut content = String::from(
+        "type IdentityWithParam<TParam> = <TResolverReturn>(
+    x: (param: TParam) => TResolverReturn
+) => (param: TParam) => TResolverReturn;
+
+type WhitespaceCharacter = ' ' | '\\n';
+type Whitespace<In> = In extends `${WhitespaceCharacter}${infer In}`
+? Whitespace<In>
+: In;
+
+type MatchesWhitespaceAndString<
+TString extends string,
+T
+> = Whitespace<T> extends `${TString}${string}` ? T : never;",
```

Comment:
let's add some line breaks here, otherwise it shows up as

```
> = Whitespace<T> extends `${TString}${string}` ? T : never;export function iso<T>(
```